internal/util/decompressor.go

```diff
@@ -179,21 +180,60 @@ type ZipDecompressor struct {
 }
 
 func (z *ZipDecompressor) Decompress(dest string) error {
+	rootFolderInZip := FindRootFolderInZip(z.src)
 	r, err := zip.OpenReader(z.src)
 	if err != nil {
 		return err
 	}
 	defer r.Close()
 	for _, f := range r.File {
-		err := z.processZipFile(f, dest)
+		err := z.processZipFile(f, dest, rootFolderInZip)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (z *ZipDecompressor) processZipFile(f *zip.File, dest string) error {
+func FindRootFolderInZip(zipFilePath string) string {
+	r, err := zip.OpenReader(zipFilePath)
+	if err != nil {
+		return ""
+	}
+	defer r.Close()
+
+	// Stores the first element of the path of all files
+	var firstElements []string
+
+	// Checks if the slice contains the specified element
+	contains := func(slice []string, element string) bool {
```

Comment:
We can use `map` to replace this algorithm that the time complexity is a bit high

Reply:
Okay, I've optimized the function